### PR TITLE
chore: test all supported ESLint versions

### PR DESCRIPTION
**Pull request by MichaelDeBoey**
As asked by @btmills in https://github.com/eslint/eslint-plugin-markdown/pull/195#issuecomment-955802046

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -27,23 +27,33 @@ jobs:
     strategy:
       matrix:
         os: [ubuntu-latest]
-        node-version: ["8.10.0", 8.x, 10.x, 12.x, 13.x, 14.x]
+        eslint: [6, 7]
+        node: [8.10, 8, 10, 12, 14, 16]
```
Comment (btmills) [inline .github/workflows/ci.yml:31]: This ought to fix the CI failure:

```suggestion
        node: [“8.10.0”, 8, 10, 12, 14, 16]
```

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -27,23 +27,33 @@ jobs:
...
       matrix:
         os: [ubuntu-latest]
-        node-version: ["8.10.0", 8.x, 10.x, 12.x, 13.x, 14.x]
+        eslint: [6, 7]
+        node: ['8.10.0', 8, 10, 12, 14, 16]
+        exclude:
+          - eslint: 7
+            node: 8
+          - eslint: 7
+            node: 8.10
```
Comment (btmills) [inline .github/workflows/ci.yml:36]: Sorry, I missed one occurrence:

```suggestion
            node: “8.10.0”
```

**Comment by nzakas** [comment]
Looks like there may be an issue on Mac CI.

**Comment by MichaelDeBoey** [comment]
@nzakas all tests are green

**Review by btmills — APPROVED** [review]
LGTM, thanks @MichaelDeBoey! I'll give others another day to review before merging.

**Review by nzakas — APPROVED** [review]

**Pull request merged**